docs/content/en/frontend/storefront-api.md

```diff
@@ -27,435 +27,202 @@ php artisan vendor:publish --tag="shopify-theme"
-  const variantId = form.querySelector('#ss-product-variant')
-
-  if (variantId == null) {
-    return
-  }
-
-  const lineItemsToAdd = [
-    {
-      variantId: variantId.value,
-      quantity: quantity != null ? parseInt(quantity.value) : 1
-    }
-  ]
-
-  client.checkout
-    .addLineItems(checkoutId, lineItemsToAdd)
-    .then(checkout => {
-      const elements = htmlToElements(
-        '<p><span class="mr-2">Product added to the basket.</span><a href="/cart">Go to cart</a></p>'
-      )
-      bannerMessage(elements, true)
-      setCartCount(checkout.lineItems)
-    })
-    .catch(err => {
-      // Handle Errors here.
-    })
-}
-```
-
-<alert scope="info">
-
-We use two functions `bannerMessage()` and `setCartCount()` which we will address further down the page. In short, these display feedback to the user.
-
-</alert>
-
-## Cart.js
-
-This is the big one. We handle pulling the cart details into our site, as well as updating quantity, managing deletion of products and handing off to Shopify.
-
-### setCartCount
-
-In the demo theme, we add a count into the header so the user can see how many products are in the basket. This function returns the cart, checks the total quantity and sets the count to that value.
 
+### getExistingCart
 ```js
-const setCartCount = () => {
-  const countTarget = document.querySelector('[data-ss-cart-count]')
-
-  if (countTarget == null) {
-    return
-  }
-
-  client.checkout
-    .fetch(checkoutId)
-    .then(({ lineItems }) => {
-      let count = 0
-      lineItems.forEach(item => (count = count + item.quantity))
-      countTarget.innerHTML = count
-    })
-    .catch(err => {
-      // Handle Errors here.
-    })
-}
+getExistingCart(string: cartId)
 ```
+Returns the cart identified by the passed id.
 
-### hideCartOverview
-
-Toggles whether the cart should be displayed. This is used upon loading the cart page.
 
+### getOrCreateCart
 ```js
-const hideCartOverview = () => {
-  noItemsMessage.classList.remove('hidden')
-  cartView.classList.add('hidden')
-}
+getOrCreateCart(?string: cartId)
 ```
+Gets a cart that matches the id. If one cannot be found, then a new cart will be returned instead.
 
-### setCartSubtotal
-
-Updates the subtotal count with the `amount` passed to it. In our case, we pass the subTotal of the whole checkout.
 
+### setCartAttributes
 ```js
-const setCartSubtotal = amount => {
-  const subtotalEl = document.querySelector('[data-ss-subtotal]')
-
-  if (subtotalEl != null) {
-    subtotalEl.innerHTML = formatCurrency(amount)
-  }
-}
+setCartAttributes(string: cartId, array: lines)
 ```
-
-### initCartActions
-
-Whenever we change or append rows to the table we need to reinitialise our actions so that we can update the product quantity or delete the product entirely. We search for all `tr` in the `#ss-cart-view` table.
+Set attributes associated with the cart. This should be an array of objects consisting of a `key` and `value`. Eg:
```

Comment:
```suggestion
Set attributes associated with the cart. This will overwrite previously set attributes if you have any. The data should be an array of objects consisting of a `key` and `value`. Eg:
```

Would be good to expressly state they'll be overwritten, otherwise it might be interpreted as it tacks them onto ones previously set :)

Reply:
Thanks